### ISSUE TITLE
Fix the UX of thread composer to prevent accidental community/channel posts

### ISSUE THREAD

**Issue by brianlovin** [withspectrum/spectrum#2872]
The thread composer today makes it ridiculously easy to accidentally post in the wrong community or channel; we need to make this way more clear or make community/channel selection a discrete step in the thread publishing flow.

**Comment by mxstbr** [comment]
I think a quick win here would be:

- Don't select any community or channel by default, leave those empty and grey out the publish button until both are selected
- Move the selects down next to the publish button
- Only on the community view and channel views select a community and channel by default

**Comment by mxstbr** [comment]
Yep that's it @bishwenduk029!

**Comment by dev-drprasad** [comment]
@bishwenduk029 

>  I am assuming the thread composer is the New Conversion UI wherein "To:" field will remain empty
we don't need show "To:" also, put some title like "New Conversation" what do you say @mxstbr 

If you need any help, i can (though i also new)

**Comment by brianlovin** [comment]
Yep! This will also affect some layout stuff on mobile, but we'll probably want to at least match behavior where the community/channel selection requires the user to make the decision before hitting publish :)

**Comment by bishwenduk029** [comment]
Hey @dev-drprasad I was working on this issue. I was able to make changes, but the onChange event is not firing on select....have you come across such an issue before ? RequiredSelector in ComposerWithData....I tried but was unable to post this question on Spectrum.Chat so posting here ....

**Comment by dev-drprasad** [comment]
@bishwenduk029 since we are moving `RequiredSelector` to other position in same component `onChange` should work. can check browser console for any errors ??

**Comment by bishwenduk029** [comment]
In local/dev env I think there is some issue with select, onChange event is not being fired even with alpha branch code. Has anyone faced a similar problem, or am i doing something wrong ? Should I create a pull request for review? I have completed the changes but unable to test them because of select...

**Comment by dev-drprasad** [comment]
@bishwenduk029 please send PR, so that I can take a look

**Comment by dev-drprasad** [comment]
@bishwenduk029 if you have only **spectrum** community in you local development, you can't  trigger `onChange` because there is no other community to change. try creating one more community in local development.

**Comment by bishwenduk029** [comment]
Hi @dev-drprasad I have created a pull request for the preliminary changes and was hoping if you could review it and let me know if I am going in the right path, meanwhile I will look perform some analysis to find if any component may need change, also I think I have to modify the cypress scripts.
